### ISSUE TITLE
verify that all Checks present as web site 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#783]
update verify stage of maven build by special script that to ensure that all Check are referenced at that exists at Checkstyle ("*Check.java") exists at page https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/checks.xml , is some check is missed maven build should fail.

**Comment by mkordas** [comment]
I can do that, @romani please assign me.

I have working prototype as normal JUnit test. It found that [OverloadMethodsDeclarationOrderCheck](../blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java) is not referenced on the page, so the test would fail right now.

Moreover, we have the following classes named `*Check.java` while not being real checks. What should we do with them? Maybe we should move them to separate package, away of normal checks?
```
AbstractFileSetCheck
FileSetCheck
AbstractFormatCheck
AbstractOptionCheck
AbstractTypeAwareCheck
AbstractIllegalCheck
AbstractIllegalMethodCheck
AbstractNestedDepthCheck
AbstractSuperCheck
AbstractHeaderCheck
AbstractJavadocCheck
AbstractClassCouplingCheck
AbstractComplexityCheck
AbstractAccessControlNameCheck
AbstractNameCheck
AbstractTypeParameterNameCheck
AbstractParenPadCheck
```

**Comment by romani** [comment]
resolved, will be in 6.5
